packages/ariakit/src/checkbox/__examples__/checkbox-as-button/index.tsx

```diff
@@ -0,0 +1,27 @@
+import { useState } from "react";
+import { Checkbox, useCheckboxState } from "ariakit/checkbox";
+import { VisuallyHidden } from "ariakit/visually-hidden";
+import "./style.css";
+
+export default function Example() {
+  const checkbox = useCheckboxState();
+  const [focusVisible, setFocusVisible] = useState(false);
+  return (
+    <label className="label">
+      <VisuallyHidden>
+        <Checkbox
+          state={checkbox}
+          onFocusVisible={() => setFocusVisible(true)}
+          onBlur={() => setFocusVisible(false)}
+        />
+      </VisuallyHidden>
+      <div
+        className="checkbox"
+        data-focus-visible={focusVisible ? "" : undefined}
+        data-checked={checkbox.value}
+      >
+        {checkbox.value ? "Checked" : "Unchecked"}
+      </div>
+    </label>
+  );
+}
```

Comment:
Now that it looks like a button, we can use `<Checkbox as="button">` 😄

I also added the `clickOnEnter` prop because that makes more sense on non-native checkboxes (maybe we should set it to true by default when it's not a native checkbox?).

```suggestion
import { Checkbox, useCheckboxState } from "ariakit/checkbox";
import "./style.css";

export default function Example() {
  const checkbox = useCheckboxState();
  return (
    <Checkbox
      as="button"
      state={checkbox}
      className="checkbox"
      clickOnEnter
    >
      {checkbox.value ? "Checked" : "Unchecked"}
    </Checkbox>
  );
}
```

Reply:
Haha 🤪 This is sort of what I was trying to ask about in my other comment. I was confused when to use `as=` or not.  Since it is a button, is the assumption that it shouldn't be used in a form because it no longer has a native `checkbox`?

Reply:
Yeah! When adding documentation to that example, we should recommend against using a button like that inside a form.

Reply:
Yeah I think the `clickOnEnter` makes sense here. Weirdly trying to treat it like a checkbox and a button.